### PR TITLE
change to package_operator so that contrib - delivery sync dag works

### PR DESCRIPTION
**Pull request by mackeynichols**

### CODE REVIEW

**Comment by mackeynichols** [comment]
[Airflow Job to make sure PROD Contrib and Delivery Package Lists are the same](https://trello.com/c/CVJXWOjT/84-airflow-job-to-make-sure-prod-contrib-and-delivery-package-lists-are-the-same)

File: dags/sustainment/check_contrib_delivery_sync.py [hunk 1]
```diff
@@ -18,10 +18,6 @@
 
 from utils import airflow_utils
 
-
```
Comment (mackeynichols) [inline dags/sustainment/check_contrib_delivery_sync.py:21]: We dont need these credentials, so Im removing them

File: dags/sustainment/check_contrib_delivery_sync.py [hunk 2]
```diff
@@ -33,9 +29,9 @@
         "email_on_failure": False,
         "email_on_retry": False,
         "retries": 1,
-        "on_failure_callback": task_failure_slack_alert,
```
Comment (mackeynichols) [inline dags/sustainment/check_contrib_delivery_sync.py:36]: This failure alert is redundant - when a task in this fails, there's already an alert sent to slack

**Comment by careduz** [comment]
Hey - this isn't needed any longer since you figured out the discrepancy with the API call results, correct?

**Comment by mackeynichols** [comment]
@careduz re https://github.com/open-data-toronto/operations/pull/33#issuecomment-911861341

Good question.
IMO It is still needed. If the SOLR issue we're currently facing appears again, this fix will find it and identify it accurately.

Without this fix, this DAG will not properly identify the cause of the issue.

Let me know if you want me to explain further

File: plugins/ckan_operators/package_operator.py [hunk 3]
```diff
@@ -31,14 +31,32 @@ def __init__(self, address, apikey, **kwargs):
         self.ckan = ckanapi.RemoteCKAN(apikey=apikey, address=address)
 
     def execute(self, context):
-        packages = self.ckan.action.package_list()
+        # Initiate a loop of getting each package from CKAN, 1000 packages at a time
+        output = []
+        offset = 0
 
-        return {"packages": self.ckan.action.package_search(rows=len(packages))["results"]}
```
Comment (mackeynichols) [inline plugins/ckan_operators/package_operator.py:36]: Hey @careduz - I've replaced the 1000 hardcoded max packages to be returned with a smarter loop

On a loop:
It reads the count of packages returned after asking for 1000 and, if theres reason to believe there are more packages, it adds an offset and gets more packages

Comment (careduz) [reply plugins/ckan_operators/package_operator.py:36]: Great, thank you!

**Review by careduz — APPROVED** [review]

**Pull request merged**